eip-4626.md

```diff
@@ -0,0 +1,158 @@
+
+Returns the total amount underlying tokens held in the vault for `_owner`.
+
+#### underlying
+`function underlying() public view returns (address)`
+
+Returns the address of the token the vault uses for accounting, depositing, and withdrawing. 
+
+SHOULD return a token implementing the ERC-20 standard.
+
+
+#### totalSupply
+
+`function totalSupply() public view returns (uint256)`
+
+Returns the total number of unredeemed vault shares in circulation.
+
+OPTIONAL - This method is only needed for vaults that implement a pro-rata share mechanism for deposits.
+
+#### balanceOf
+
+`function balanceOf(address _owner) public view returns (uint256)`
+
+Returns the total amount of vault shares the `_owner` currently has.
+
+OPTIONAL - This method is only needed for vaults that implement a pro-rata share mechanism for deposits.
+
+#### redeem
+
+`function redeem(address _to, uint256 _shares) public returns (uint256 _value)`
+
+Redeems a specific number of `_shares` for underlying tokens and transfers them to `_to`. 
+
+MAY return a pro-rata ownership `_shares` value corresponding `_value`, if not MUST return `0`.
+
+OPTIONAL - This method is only needed for vaults that implement a pro-rata share mechanism for deposits.
+
+#### exchangeRate
+
+`function exchangeRate() public view returns (uint256)`
+
+The amount of underlying tokens one `baseUnit` of vault shares is redeemable for. 
+
+e.g. `_shares * exchangeRate() / baseUnit() = _value`.
+
+`exchangeRate() * totalSupply()` MUST equal `totalHoldings()`.
+
+OPTIONAL - This method is only needed for vaults that implement a pro-rata share mechanism for deposits.
+
+#### baseUnit
+
+`function baseUnit() public view returns(uint256)`
+
+The decimal scalar for vault shares and operations involving `exchangeRate()`.
+
+OPTIONAL - This method is only needed for vaults that implement a pro-rata share mechanism for deposits.
+
+
+### Events
+
+#### Deposit
+
+MUST be emitted when tokens are deposited into the vault.
+
+`event Deposit(address indexed _from, addres indexed _to, uint256 _value)`
+
+Where `_from` is the user who triggered the deposit and approved `_value` underlying tokens to the vault, and `_to` is the user who is able to withdraw the deposited tokens. 
+
+
+#### Withdraw
+
+MUST be emitted when tokens are withdrawn from the vault by a depositor.
+
+`event Withdraw(address indexed _owner, addres indexed _to, uint256 _value)`
+
+Where `_from` is the user who triggered the withdrawal and held `_value` underlying tokens in the vault, and `_to` is the user who received the withdrawn tokens.
+
+
+## Rationale
+
+The vault interface is designed to be optimized for minimal implementation and integration logic while maintaining flexibility for both parties. Details such as accounting and allocation of deposited tokens are intentionally not specified, as vaults are expected to be treated as black boxes on-chain and inspected off-chain before use.
+
```

Comment:
There should be a "Backwards Compatability" section here, I think (someone will correct me if it isn't strictly necessary - even if it is, I would recommend spending a paragraph on what this means for platforms that are not following this EIP)

Reply:
We elected not to add one because its a pretty standalone standard and there isn't much to say. Its backwards compatible with ERC20 and all implementations which don't use this interface should either update or lose out on the network effects

Reply:
Backwards Compatibility is a required section per EIP-1, so you will need to include something short even if it's just stating that there are no compatibility issues.